### PR TITLE
ADBDEV-6459 Fix -Wunused-variable and -Wunused-but-set-variable warnings produced by Clang

### PR DESCRIPTION
**Pull request by dnskvlnk**
Fix -Wunused-variable and -Wunused-but-set-variable warnings produced by Clang

### CODE REVIEW

**Comment by andr-sokolov** [comment]
The description contains two identical sentences. They can be combined in one "Fix  -Wunused-variable and -Wunused-but-set-variable warnings produced by Clang". The first letter in the Clang word should be in upper case

Typo:  -Wunsued-variable

File: src/backend/postmaster/pgstat.c [hunk 1]
```diff
@@ -4820,17 +4820,11 @@ pgstat_combine_one_qe_result(List **oidList, struct pg_result *pgresult,
...
 	PgStat_TableStatus	   *pgstat_info;
 	PgStat_TableXactStatus *trans;
 
 	if (!pgresult || pgresult->extraslen < 1 || pgresult->extraType != PGExtraTypeTableStats)
 		return;
-	/*
-	* If this is the first rel to be modified at the current nest level,
-	* we first have to push a transaction stack entry.
-	*/
-	xact_state = get_tabstat_stack_level(nest_level);
```
Comment (andr-sokolov) [inline src/backend/postmaster/pgstat.c:4833]: Why is it correct to remove the get_tabstat_stack_level function call?

**Comment by bandetto** [comment]
Please change `-Wunsued-variable` to `-Wunused-variable` in title and PR description.

File: src/backend/postmaster/pgstat.c (hunk 1 above)
Comment (bandetto) [reply src/backend/postmaster/pgstat.c:4833]: `get_tabstat_stack_level()` call might modify `pgStatXactStack` global variable.

Comment (dnskvlnk) [reply src/backend/postmaster/pgstat.c:4833]: Yes, I agree, it would be better just to discard function's result

Comment (dnskvlnk) [reply src/backend/postmaster/pgstat.c:4833]: I reverted this change

**Review by andr-sokolov — APPROVED** [review]

**Review by bandetto — APPROVED** [review]

**Pull request merged**